bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/internal/SecureHttpClientFactory.java

```diff
@@ -0,0 +1,278 @@
+
+    private static final String CONFIG_MIN_THREADS_SHARED = "minThreadsShared";
+    private static final String CONFIG_MAX_THREADS_SHARED = "maxThreadsShared";
+    private static final String CONFIG_KEEP_ALIVE_SHARED = "keepAliveTimeoutShared";
+    private static final String CONFIG_MIN_THREADS_CUSTOM = "minThreadsCustom";
+    private static final String CONFIG_MAX_THREADS_CUSTOM = "maxThreadsCustom";
+    private static final String CONFIG_KEEP_ALIVE_CUSTOM = "keepAliveTimeoutCustom";
+
+    private static final int MIN_CONSUMER_NAME_LENGTH = 4;
+    private static final int MAX_CONSUMER_NAME_LENGTH = 20;
+    private static final Pattern CONSUMER_NAME_PATTERN = Pattern.compile("[a-zA-Z0-9_\\-]*");
+
+    private volatile TrustManagerProvider trustmanagerProvider;
+
+    private QueuedThreadPool threadPool = null;
+    private HttpClient commonHttpClient = null;
+    private CommonHttpClient commonHttpClientFacade = null;
+
+    private int minThreadsShared;
+    private int maxThreadsShared;
+    private int keepAliveTimeoutShared;
+    private int minThreadsCustom;
+    private int maxThreadsCustom;
+    private int keepAliveTimeoutCustom;
+
+    @Activate
+    protected void activate(Map<String, Object> parameters) {
+        getConfigParameters(parameters);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> parameters) {
+        getConfigParameters(parameters);
+        if (threadPool != null) {
+            threadPool.setMinThreads(minThreadsShared);
+            threadPool.setMaxThreads(maxThreadsShared);
+            threadPool.setIdleTimeout(keepAliveTimeoutShared * 1000);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        try {
+            if (commonHttpClient != null) {
+                commonHttpClient.stop();
+                commonHttpClientFacade = null;
+                commonHttpClient = null;
+                threadPool = null;
+                logger.info("jetty shared http client stopped");
+            }
+        } catch (Exception e) {
+            logger.error("error while stopping jetty shared http client", e);
+            // nothing else we can do here
+        }
+
+    }
+
+    @Override
+    public HttpClient createHttpClient(@NonNull String consumerName, @NonNull String endpoint) {
+        Objects.requireNonNull(endpoint, "endpoint must not be null");
+        logger.debug("httpClient for endpoint {} requested", endpoint);
+        checkConsumerName(consumerName);
+        return createHttpClientInternal(consumerName, endpoint);
+    }
+
+    @Override
+    public CommonHttpClient getCommonHttpClient() {
+        initialize();
+        logger.debug("shared httpClient requested");
+        return commonHttpClientFacade;
+    }
+
+    private void getConfigParameters(Map<String, Object> parameters) {
+        minThreadsShared = getConfigParameter(parameters, CONFIG_MIN_THREADS_SHARED, 10);
```

Comment:
I think constants would in general be a better maintainable place to store the default values for such parameters.

Reply:
The defaults are defined in the metatype xml definitions. These are last resort values, if something with the configadmin goes terribly wrong. Defining constants would be misleading, because they are normally not used.